src/robot/libraries/String.py

```diff
@@ -237,29 +237,38 @@ def get_lines_matching_pattern(self, string, pattern, case_insensitive=False):
             matches = lambda line: fnmatchcase(line, pattern)
         return self._get_matching_lines(string, matches)
 
-    def get_lines_matching_regexp(self, string, pattern):
+    def get_lines_matching_regexp(self, string, pattern, exact_match=True):
         """Returns lines of the given ``string`` that match the regexp ``pattern``.
 
         See `BuiltIn.Should Match Regexp` for more information about
         Python regular expression syntax in general and how to use it
-        in Robot Framework test data in particular. A line matches
-        only if it matches the ``pattern`` fully. Notice that to make
-        the match case-insensitive, you need to embed case-insensitive
-        flag into the pattern.
-
-        Lines are returned as one string catenated back together with
+        in Robot Framework test data in particular. By default ``exact_match``
+        is True causing line matches only if it matches the ``pattern`` fully.
+        When passing ``exact_match`` as False, this keyword uses the matching
+        pattern without the '^' and '$' positional markers, searching content
+        within the input lines. Notice that to make the match case-insensitive,
+        you need to embed case-insensitive flag into the pattern.
+
+        Lines are returned as one string concatenated back together with
         newlines. Possible trailing newline is never returned. The
         number of matching lines is automatically logged.
 
         Examples:
-        | ${lines} = | Get Lines Matching Regexp | ${result} | Reg\\\\w{3} example |
-        | ${ret} = | Get Lines Matching Regexp | ${ret} | (?i)FAIL: .* |
-
-        See `Get Lines Matching Pattern` and `Get Lines Containing String` if
-        you do not need full regular expression powers (and complexity).
+        | ${lines} = | Get Lines Matching Regexp | ${result} | Reg\\\\w{3} example |   |
+        | ${ret} = | Get Lines Matching Regexp | ${ret} | (?i)FAIL: .* |   |
+        | ${result} = | Get Lines Matching Regexp | ${input} | ^ABC | False |
+        | ${result} = | Get Lines Matching Regexp | ${input} | ^ABC.* | True |
+
+        See `Get Lines Matching Pattern` and `Get Lines Containing
+        String` if you do not need full regular expression powers (and
+        complexity).
```

Comment:
A note about `exact_match` being new in 2.9 needs to added. You could explain it where `exact_match` is explained in general, but it might be easier just to add a note like this to the end of the docstring:

> `exact_match` argument is new in Robot Framework 2.9. In earlier versions exact match was always required.